### DISCUSSION TITLE
LDAP load groups recursively: User has parent group but parent group does not have user

### DISCUSSION THREAD

**Discussion by jhollmannk** [keycloak/keycloak#24503] [Q&A]
We want to use the LOAD_GROUPS_BY_MEMBER_ATTRIBUTE_RECURSIVELY setting for an ActiveDirectory LDAP.

Preserve Group Inheritance is Off as we have subgoups that are members of several parent groups.

After importing the groups we have the effect that a user that is listed in a subgroup has the subgroup itself and all parent groups in the "Membership" column:

![grafik](https://github.com/keycloak/keycloak/assets/18438281/a624fc98-c726-40f3-bea4-c24b27efe03c)

But the parent groups (like admins) do not have the user as a member:

![grafik](https://github.com/keycloak/keycloak/assets/18438281/34c01341-de12-4b61-ab37-abd8e39704e9)

The group is completely empty as it only contains subgroups (no directly assigned users).

So if we get the groups of a user all subgroups and parent groups are returned which is good.
But if we get the members of a parent group the users of a subgroup are not listed.

Is that intentional? Or are we missing a settings? It is quite confusing and we rely on getting the users of a group (not the other way around) for some functionality.

**Comment by Jerio75** [comment]
Hello 

any updates on this question ?

i just encountered the same issue, can't load groups by member recursively in our AD, with a query ldapsearch i can see clearly the team members inside the group but can't fetch at all on Keycloak, the group members (like in the screenshot) are empty
